pkg/kubectl/rolling_updater.go

```diff
@@ -353,49 +404,101 @@ func (r *RollingUpdater) Update(config *RollingUpdaterConfig) error {
-			return err
+	}
+	// Helpful output about what we're about to do.
+	direction := "up"
+	if !scaleUpFirst {
+		direction = "down"
+	}
+	fmt.Fprintf(out, "Scaling up %s from %d to %d, scaling down %s from %d to 0 (scale %s first by %d each interval)\n",
+		newName, newSpec.Replicas, desired, oldName, oldSpec.Replicas, direction, scaleAmount)
+
+	// Scale newRc and oldRc until newRc has the desired number of replicas and
+	// oldRc has 0 replicas.
+	for newSpec.Replicas != desired || oldSpec.Replicas != 0 {
+		// Set up a function for scaling up.
+		scaleUp := func() error {
+			// Nothing to do.
+			if newSpec.Replicas == desired {
+				return nil
+			}
+			// Scale up new. Account for fencepost conditions. If the old spec is
+			// already scaled to 0, go ahead and scale all the way up.
+			newSpec.Replicas += scaleAmount
+			if newSpec.Replicas > desired || oldSpec.Replicas == 0 {
+				newSpec.Replicas = desired
+			}
+			fmt.Fprintf(out, "Scaling %s up to %d\n", newName, newSpec.Replicas)
+			newRc, err = r.scaleAndWait(newRc, retry, waitForReplicas)
+			if err != nil {
+				return err
+			}
+			newSpec = &newRc.Spec
+			// Perform acceptance checking, if necessary.
+			if config.UpdateAcceptor != nil {
+				fmt.Fprintf(out, "Performing acceptance check of %s\n", newName)
+				err := config.UpdateAcceptor.Accept(newRc)
+				if err != nil {
+					return fmt.Errorf("Update rejected for %s: %v", newName, err)
+				}
+			}
+			return nil
 		}
-		fmt.Printf("At end of loop: %s replicas: %d, %s replicas: %d\n",
-			oldName, oldRc.Spec.Replicas,
-			newName, newRc.Spec.Replicas)
-	}
-	// delete remaining replicas on oldRc
-	if oldRc.Spec.Replicas != 0 {
-		fmt.Fprintf(out, "Stopping %s replicas: %d -> %d\n",
-			oldName, oldRc.Spec.Replicas, 0)
-		oldRc.Spec.Replicas = 0
-		oldRc, err = r.scaleAndWait(oldRc, retry, waitForReplicas)
-		// oldRc, err = r.scaleAndWait(oldRc, interval, timeout)
-		if err != nil {
-			return err
+
+		// Set up a function for scaling down.
+		scaleDown := func() error {
+			if oldSpec.Replicas == 0 {
+				// Nothing to do.
+				return nil
+			}
+			// Scale down old. Account for fencepost conditions. If the new spec is
+			// already at desired, go ahead and scale all the way down.
+			oldSpec.Replicas -= scaleAmount
+			if oldSpec.Replicas < 0 || newSpec.Replicas == desired {
+				oldSpec.Replicas = 0
+			}
+			fmt.Fprintf(out, "Scaling %s down to %d\n", oldName, oldSpec.Replicas)
+			oldRc, err = r.scaleAndWait(oldRc, retry, waitForReplicas)
+			if err != nil {
+				return err
+			}
+			oldSpec = &oldRc.Spec
+			return nil
 		}
-	}
-	// add remaining replicas on newRc
-	if newRc.Spec.Replicas != desired {
-		fmt.Fprintf(out, "Scaling %s replicas: %d -> %d\n",
-			newName, newRc.Spec.Replicas, desired)
-		newRc.Spec.Replicas = desired
-		newRc, err = r.scaleAndWait(newRc, retry, waitForReplicas)
-		if err != nil {
-			return err
+
+		// Perform the scaling interval in the specified direction.
+		if scaleUpFirst {
```

Comment:
The difference between these two directions is `up, down, up, down` or `down, up, down, up`.  If ups and downs are no-ops when at the limits (which they should be), then you can do this loop like:

```
if !skipUpOnFirstLoop {
  ... scale up
  time.Sleep()
  skipUpOnFirstLoop = false
}
... scale down
```